src/Data/ExistsRE.purs

```diff
@@ -0,0 +1,20 @@
+module Data.ExistsRE where
+
+import Unsafe.Coerce (unsafeCoerce)
+
+-- | A variant of `Exists` that works for type constructors that accept a _row_
+-- | of effects.
+data ExistsRE (f :: # ! -> *)
```

Comment:
Interesting. How is it different if we define a type with constructor but then hide the constructor export then? 

I'm actually not entirely sure why it's defined as this empty type, by the way. I think something like this would work too:
```
newtype ExistsRE (f :: # ! -> *) = ExistsRE (forall eff. F f eff)
```
I just copied `Exists`/`ExistsR` and tweaked the kind.